test/parser/statement/Function.test.js

```diff
@@ -178,6 +178,69 @@ describe("parser", () => {
             expect(statements).not.toBeNull();
             expect(statements).toMatchSnapshot();
         });
+
+        it('does not allow type designators at end of name', () => {
+            const { tokens } = brs.lexer.Lexer.scan(`
+                function StringFunc#()
+                    return 1
+                end function
+
+                function IntegerFunc%()
+                    return 1
+                end function
+
+                function FloatFunc!()
+                    return 1
+                end function
+
+                function DoubleFunc#()
+                    return 1
+                end function
+            `);
+            const { statements, errors } = parser.parse(tokens);
+            expect(errors.length).toEqual(4);
+            expect(errors[0].location).toMatchObject({
```

Comment:
Rather than building these locations manually, consider using `expect(errors).toMatchSnapshot()`!  Please correct me if I'm wrong, but it seems like the spirit of these tests is to make sure four errors are produced, not necessarily the locations themselves.  Since we probably want the tests to fail if the messages change too, using a snapshot might save some space and make future updates to these tests easier!

Reply:
So I _did_ want to verify the errors were in the proper locations (i.e. are these the errors I was actually looking for). However, since my initial test passed, I can assume the snapshot is also valid, so I think you're right, just use another snapshot.